File: CPP/01 Just to start/node_with_class.cpp
/**
 * @file node_with_class.cpp
 *
 * @brief A basic ROS2 node implemented with class that spins
 *        and printouts a string waiting for a killing request
 *
 * @author Antonio Mauro Galiano
 * Contact: https://www.linkedin.com/in/antoniomaurogaliano/
 *
 */


#include "rclcpp/rclcpp.hpp"


class MyNode : public rclcpp::Node    // class declaration which inherits from rclcpp::Node
{
public:
  MyNode() : Node("ros2_node_class")
  {
    RCLCPP_INFO_STREAM(this->get_logger(), "Echo line from a ROS2 class node");   // just a printout
  }
};


int main(int argc, char **argv) {
  rclcpp::init(argc, argv);
  auto node = std::make_shared<MyNode>();   // declaration of a MyNode class node
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}

File: PYTHON/start_with_simple_nodes_py/start_with_simple_nodes_py/node_with_class.py
#  *
#  * @file node_with_class.py
#  *
#  * @brief A basic ROS2 node implemented with class that spins
#  *        and printouts a string waiting for a killing request
#  *
#  * @author Antonio Mauro Galiano
#  * Contact: https://www.linkedin.com/in/antoniomaurogaliano/
#  *
#  *

import rclpy
from rclpy.node import Node

class MyFirstNode(Node):
    def __init__(self):
        super().__init__('my_first_node_class')
        self.get_logger().info("ROS2 class node first test")

def main(args=None):
    rclpy.init(args=args)
    node = MyFirstNode()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        if rclpy.ok():
            node.get_logger().info("Shutting down node...")
    finally:
        node.destroy_node()
        if rclpy.ok():
            rclpy.shutdown()

if __name__ == '__main__':
    main()